src/templates/study-space.js

```diff
@@ -19,10 +20,32 @@ const StudyTemplate = ({ data }) => {
       ?.localFile?.childImageSharp?.gatsbyImageData;
   const imageAlt = relationships?.field_media_image?.field_media_image?.alt || '';
 
+  const location = useLocation();
+  const fromFindStudySpace = location.state && location.state.fromFindStudySpace;
+  const findStudySpaceQuery = location.state && location.state.findStudySpaceQuery;
+  const findStudySpaceURL = '/visit-and-study/study-spaces/find-study-space';
+
   return (
     <TemplateLayout node={node}>
       <Margins>
         <Breadcrumb data={fields.breadcrumb} />
+        {fromFindStudySpace
+          ? (
+              <div
+                style={{
+                  marginBottom: SPACING.XL
+                }}
+              >
+                <Link
+                  onClick={() => {
+                    navigate(`${findStudySpaceURL}${fromFindStudySpace ? findStudySpaceQuery : '' || ''}`);
```

Comment:
Could `findStudySpaceQuery` potentially return `undefined`? If so, maybe `line 25` can be rewritten as such:
```
const findStudySpaceQuery = (location.state && location.state.findStudySpaceQuery) || '';
```

This way you won't need the nested ternary operator.